### PR TITLE
add Constructor Validate and Exception.

### PR DESCRIPTION
**Pull request by LoicGady-Exa**

### CODE REVIEW

File: tests/ValidateExceptionTest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
```
Comment (bouiboui) [inline tests/ValidateExceptionTest.php]: This test has no use (except coverage 😋).

File: tests/ValidateTest.php [hunk 2]
```diff
@@ -40,7 +43,7 @@ public function testGetRegexAsString()
     public function testValidateEmailFormatInvalidAsStringReturn()
     {
         $this->assertEquals(
-            Exaprint\MailHelper\Validate::ERROR, Exaprint\MailHelper\Validate::validateEmailFormat('error')
```
Comment (bouiboui) [inline tests/ValidateTest.php]: You should `import Exaprint\MailHelper\Validate` to make your code less verbose.

File: tests/ValidateTest.php [hunk 3]
```diff
@@ -50,7 +53,7 @@ public function testValidateEmailFormatInvalidAsStringReturn()
     public function testValidateEmailFormatValidAsNullReturn()
     {
         $this->assertEquals(
-            null, Exaprint\MailHelper\Validate::validateEmailFormat('david.jones@proseware.com')
+            null, Exaprint\MailHelper\Validate::validateEmailFormat(self::validMail)
         );
```
Comment (bouiboui) [inline tests/ValidateTest.php]: I think there is a `assertNull`.

File: tests/ValidateTest.php [hunk 4]
```diff
@@ -78,4 +81,114 @@ public function testValidateEmailFormat()
...
+
+    /**
+     * @covers Exaprint\MailHelper\Validate::validate()
+     */
+    public function testValidateWithoutException()
+    {
+        $o = Exaprint\MailHelper\Validate::getValidate();
+        $this->assertTrue($o->validate(self::validMail));
+        $this->assertFalse($o->validate(self::invalidMail));
+        $o = Exaprint\MailHelper\Validate::getValidate(true);
```
Comment (bouiboui) [inline tests/ValidateTest.php]: Avoid booleans as parameters, they make code less readable.
Now that you have a real object, you can easily implement something like:
```php
$o = (Validate::getValidate())->setThrowException(true);
```

File: tests/ValidateExceptionTest.php (hunk 1 above)
Comment (LoicGady-Exa) [inline tests/ValidateExceptionTest.php]: ?

File: tests/ValidateTest.php (hunk 4 above)
Comment (LoicGady-Exa) [inline tests/ValidateTest.php]: Prefer no use function setThrowException for use validate many times.
```PHP
$o = Validate::getValidate(true);
$o->validate($email1);
$o->validate($email2, false);
$o->validate($email3);
```
With your function :
```PHP
$o = (Validate::getValidate())->setThrowException(true);
$o->validate($email1);
$o->setThrowException(false);
$o->validate($email2, false);
$o->setThrowException(true);
$o->validate($email3);
```

File: tests/ValidateTest.php (hunk 2 above)
Comment (LoicGady-Exa) [inline tests/ValidateTest.php]: ok

File: tests/ValidateTest.php (hunk 3 above)
Comment (LoicGady-Exa) [inline tests/ValidateTest.php]: ok

File: tests/ValidateExceptionTest.php (hunk 1 above)
Comment (LoicGady-Exa) [inline tests/ValidateExceptionTest.php]: Does not need unit test on `ValidateException` class ?

Comment (bouiboui) [inline tests/ValidateExceptionTest.php]: Nope, you must test the code that throws the Exceptions, but the Exceptions themselves have no logic to test.

File: tests/ValidateTest.php (hunk 4 above)
Comment (bouiboui) [inline tests/ValidateTest.php]: I understand, would you accept this:
```php
$o = Validate::getValidate(Validate::THROW_EXCEPTIONS);
$o->validate($email1);
$o->validate($email2, Validate::DO_NOT_THROW_EXCEPTIONS);
$o->validate($email3);
```
?

File: tests/ValidateExceptionTest.php (hunk 1 above)
Comment (LoicGady-Exa) [inline tests/ValidateExceptionTest.php]: ok

File: tests/ValidateTest.php (hunk 4 above)
Comment (LoicGady-Exa) [inline tests/ValidateTest.php]: ok

File: tests/ValidateTest.php [hunk 5]
```diff
@@ -25,57 +31,163 @@
...
+        $this->expectException(ValidateException::class);
+        $o->validate();
+    }
+
+    /**
+     * @covers \Exaprint\MailHelper\Validate::validate()
+     */
+    public function testValidateWithoutException()
+    {
+        $o = Validate::getValidate();
```
Comment (bouiboui) [inline tests/ValidateTest.php]: @BasileTrujillo would tell you to split this in 5 different tests :) (best parallelism possible).
Can you add a new line before each `$o = `?

**Pull request merged**